nonfree/nvidia-container-toolkit/nvidia-persistenced-wrapper/main.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"errors"
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"os/signal"
+	"strconv"
+
+	"golang.org/x/sys/unix"
+)
+
+const (
+	stateFolder = "/var/run/nvidia-persistenced"
+	pidFile     = stateFolder + "/" + "nvidia-persistenced.pid"
+)
+
+func main() {
+	// first check if the pid file exists,
+	// then check if the process is running,
+	// if running try to kill it, then start the new process
+	if _, err := os.Stat(pidFile); err == nil {
```

Comment:
nit: can do return in Stat to avoid nested ifs. And use `log.Fatalf` instead of `log.Printf` + `os.Exit(1)`

```go
if _, err := os.Stat(pidFile); err != nil {
  if !errors.Is(err, os.ErrNotExist) {
    log.Fatalf("nvidia-persistenced-wrapper: error starting nvidia-persistenced: %v\n", err)
  }
}

if pid, err := getProcessId(); err != nil {
  log.Fatalf("nvidia-persistenced-wrapper: error reading pid file: %s%v\n", pidFile, err)
}

// and so on

Reply:
Thanks, fixed